File: 1160 - Find Words That Can Be Formed by Characters.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int CountCharacters(string[] words, string chars)
    {
        var map = new Dictionary<char, int>();
        foreach (char c in chars)
        {
            if (!map.ContainsKey(c))
            {
                map.Add(c, 1);
            }
            else
            {
                map[c]++;
            }
        }
        int res = 0;
        foreach (string word in words)
        {
            var t = new Dictionary<char, int>();
            foreach (char c in word)
            {
                if (!t.ContainsKey(c))
                {
                    t.Add(c, 1);
                }
                else
                {
                    t[c]++;
                }
            }
            bool good = true;
            foreach (var pair in t)
            {
                if (!map.ContainsKey(pair.Key))
                {
                    good = false;
                    break;
                }
                else if (map[pair.Key] < pair.Value)
                {
                    good = false;
                    break;
                }
            }
            if (good)
            {
                res += word.Length;
            }
        }
        return res;
    }
}

File: 1160 - Find Words That Can Be Formed by Characters.php
<?php


class Solution
{

    /**
     * @param String[] $words
     * @param String $chars
     * @return Integer
     */
    function countCharacters($words, $chars)
    {
        $count = array();
        $chars = str_split($chars);
        foreach ($chars as $c) {
            if (isset($count[$c])) {
                $count[$c]++;
            } else {
                $count[$c] = 1;
            }
        }

        $res = 0;
        foreach ($words as $word) {
            $t = array();
            $word = str_split($word);
            foreach ($word as $c) {
                if (isset($t[$c])) {
                    $t[$c]++;
                } else {
                    $t[$c] = 1;
                }
            }
            $good = true;
            foreach ($t as $k => $v) {
                if (!isset($count[$k])) {
                    $good = false;
                    break;
                } else if ($count[$k] < $v) {
                    $good = false;
                    break;
                }
            }
            if ($good) {
                $res += count($word);
            }
        }
        return $res;
    }
}
